packages/amplify-graphql-generation-transformer/src/utils/tools.ts

```diff
@@ -0,0 +1,73 @@
+import { TransformerContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { FieldDefinitionNode } from 'graphql';
+import { JSONSchema, generateJSONSchemaFromTypeNode } from './graphql-json-schema-type';
+
+export type Tool = {
+  toolSpec: ToolSpec;
+};
+
+export type Tools = {
+  tools: Tool[];
+};
+
+export type ToolConfig = {
+  tools: Tool[];
+  toolChoice: {
+    tool: {
+      name: string;
+    };
+  };
+};
+
+type ToolSpec = {
+  name: string;
+  description: string;
+  inputSchema: {
+    json: JSONSchema;
+  };
+};
+
+/**
+ * Creates a tool configuration for generating a response type based on a GraphQL field definition.
+ *
+ * This function generates a JSON schema from the field's type and wraps it in a tool specification.
+ * The tool can be used by AI models to generate responses that conform to the field's type structure.
+ *
+ * @param {FieldDefinitionNode} field - The GraphQL field definition node for which to create the response type tool.
+ * @param {TransformerContextProvider} ctx - The transformer context provider, which supplies necessary context for schema generation.
+ * @returns {ToolConfig} A tool configuration object containing:
+ *   - tools: An array with a single tool specification for the response type.
+ *   - toolChoice: An object specifying the name of the tool to be used.
```

Comment:
Similar to above: why have a full structure to specify a string value?

Reply:
Same answer as above -- it's the structure required for the Bedrock `/converse` payload.